### DISCUSSION TITLE
How to deploy SPA + Server Routes (API) to AWS

### DISCUSSION THREAD

**Discussion by steffenstolze** [nuxt/nuxt#16896] [Questions]
Hey there!

I've not been able to find guidance or an example how to deploy a Nuxt v3 SPA + API (server routes) to AWS.

In theory, it has to be deployed to CloudFront and S3 for the SPA and API Gateway and Lambda for the server routes.

What I did find was server routes through CloudFront + Lambda@Edge which is **not** what I need. One reason would be that Lambda@Edge is only available in us-east-1 region and also limited.

Everything else I find is either about **only** the frontend or SSR.

The Vercel guys must have figured it out since Vercel deploys to Vercel functions which is basically Lambda under the hood.

Did someone manage to do this and could give some guidance?

Thanks and BR!

**Comment by 20lives** [comment]
I am using Elastic beanstalk.
Easy deployment, probably much faster (and a bit more expensive) since not using lambda.
you can also use CloudFront with custom routes behavior to cache static files.
